In [1]:
from coffea import hist,  util
import awkward as ak
import numpy as np

In [2]:
from coffea.util import load, save

year = 2018
vfp = 'post'
path = '/Users/hongjieun/MonoTop/notebook_plotting/check_events/'
rg = 'tw'


In [48]:
#version = 43
dummy='''
if year == 2016:
    hists = load(path+'manyplot_'+str(year)+str(vfp)+'VFP_Run'+str(version)+'_'+str(rg)+'.scaled')
    print(path+'manyplot_'+str(year)+str(vfp)+'VFP_Run'+str(version)+'_'+str(rg)+'.scaled')
else:
    hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled') #Run all
    print(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')


data_hists = hists['data']
bkg_hists = hists['bkg']
sig_hists = hists['sig']
'''
lumi = {'2018':59.83,
       '2017': 41.48,
       '2016' : {'pre': 19.52,
                'post': 16.81} 
       }

dat = {'2018': {'e': 'EGamma',
                'm': 'MET',
               'g': 'EGamma'},
       '2017': {'e': 'SingleElectron',
               'm': 'MET',
               'g': 'SinglePhoton'},
      '2016': {'e': 'SingleElectron',
              'm': 'MET',
              'g': 'SinglePhoton'}}

eexc = [ 
    'sumw', 'template', 'dr_e_lj','ndcsvM', 'dr_mu_lj','cutflow',    
    'runnumber','ndflvL','ndflvM', 'recoil_wg',
    'recoilphi', 
]
exc = ['phopt', 'sumw', 'template', 'cutflow',  
       'elept', 'elephi', 'eleeta', 
       'mupt', 'muphi', 'mueta', 'phophi', 'phoeta', 
       'partWpt', 'partWpt2', 'TvsQCD']

region = 'wmcr'

## all genWeight

In [49]:
# With nlo ewk, all genWeight, apply HEM cut(file V30)
version = 30
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withnlo_allGenW_applyHEM']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

In [50]:
# Without nlo ewk, all genWeight, apply HEM cut(file V31)
version = 31
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withoutnlo_allGenW_applyHEM']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

In [66]:
# With nlo ewk, all genWeight, no HEM cut(file V51)
version = 51
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withnlo_allGenW_noHEMcut']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', region).integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

FileNotFoundError: [Errno 2] No such file or directory: '/Users/hongjieun/MonoTop/notebook_plotting/check_events/manyplot_2018_V51_tw.scaled'

## positive genWeight

In [51]:
# With nlo ewk, positive genWeight, apply HEM cut(file V44)
version = 44
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withnlo_posGenW_applyHEM']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

In [52]:
# With nlo ewk, positive genWeight, no HEM cut(file V53)
version = 53
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withnlo_posGenW_noHEMcut']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

FileNotFoundError: [Errno 2] No such file or directory: '/Users/hongjieun/MonoTop/notebook_plotting/check_events/manyplot_2018_V53_tw.scaled'

In [53]:
# Without nlo ewk, positive genWeight, apply HEM cut(file V43)
version = 43
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withoutnlo_posGenW_applyHEM']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

In [60]:
# Without nlo ewk, positive genWeight, no HEM cut(file V54)
version = 54
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withoutnlo_posGenW_noHEMcut']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

## negative genWeight

In [55]:
# With nlo ewk, negative genWeight, apply HEM cut(file V46)
version = 46
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withnlo_negGenW_applyHEMcut']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

In [56]:
# With nlo ewk, negative genWeight, no HEM cut(file V55)
version = 55
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withnlo_negGenW_noHEMcut']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

In [57]:
# Without nlo ewk, negative genWeight, apply HEM cut(file V45)
version = 45
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withoutnlo_negGenW_applyHEMcut']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

In [58]:
# Without nlo ewk, negative genWeight, no HEM cut(file V56)
version = 56
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withoutnlo_negGenW_noHEMcut']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

In [62]:
# With nlo ewk, all genWeight, remove HEM met cut on CR(file V60)
version = 60
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
compare_hists['withnlo_allGenW_removeHEMmetonCR']={
    'WJets-0J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-0J',)],
    'WJets-1J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-1J',)],
    'WJets-2J': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-2J',)],
    'WJets-Pt': bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()[('WJets-Pt',)]
                                             }

In [20]:
compare_hists

{'withoutnlo_posGenW_applyHEM': {'WJets-0J': array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
         0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
         0.]),
  'WJets-1J': array([  0.        ,  49.30109023, 199.48913614, 218.97585131,
         195.10788881, 190.39656555, 191.38508139, 175.3516278 ,
         183.42477458,  72.93726723,   3.33522218,   0.        ,
           0.        ,  15.82353438,  86.6528045 , 246.68019989,
         230.14092217, 268.62273252, 227.93695441, 213.54990668,
         228.61318349, 261.7835463 , 231.29859547, 199.86144523,
         216.38706851, 228.31452763, 210.68405257, 226.4205194 ,
         227.68974519, 204.79545911, 176.25797469, 285.52660551,
         183.11278999,  55.00674599,   0.        ]),
  'WJets-2J': array([  0.        , 125.00578193, 532.98484255, 546.81285991,
         654.43179966, 558.92631095, 558.44745659, 552.37768022,
         515.57498148, 231.65518645,   8.83340788,   9.559653

In [42]:
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'_'+str(rg)+'.scaled')
bkg_hists = hists['bkg']
bkg_hists['recoilphi'].integrate('region', 'wmcr').integrate('systematic','nominal').values()

{('WJets-2J',): array([   0.        ,    0.        ,    0.        ,    0.        ,
        2643.46183909,    0.        ,    0.        ,    0.        ,
        2493.60912319,  911.42505033,    0.        ,    0.        ,
           0.        ,    0.        ,  810.51074734,    0.        ,
           0.        ,  847.58261588,    0.        ,    0.        ,
           0.        ,    0.        ,    0.        ,  879.31609567,
           0.        ,    0.        ,  785.90156315,  693.14958159,
           0.        ,    0.        ,  794.13220096,  706.2493142 ,
         950.12401525,    0.        ,    0.        ])}

In [63]:
for cat in compare_hists.keys():
    print(cat)
    for i in range(14):
        print('%10.2f' % compare_hists[cat]['WJets-1J'][i])
    print()

withoutnlo_posGenW_applyHEM
      0.00
     49.30
    199.49
    218.98
    195.11
    190.40
    191.39
    175.35
    183.42
     72.94
      3.34
      0.00
      0.00
     15.82

withnlo_posGenW_applyHEM
      0.00
     44.61
    180.18
    199.25
    176.30
    171.74
    173.09
    158.94
    166.45
     67.32
      3.00
      0.00
      0.00
     14.43

withnlo_allGenW_applyHEM
      0.00
     47.92
    219.20
    212.20
    230.83
    217.36
    224.89
    165.96
    212.54
    104.71
     -0.82
      0.00
      0.00
     16.90

withoutnlo_allGenW_applyHEM
      0.00
     75.61
    436.48
    391.43
    434.18
    404.17
    370.78
    313.93
    409.70
    154.60
     17.24
     42.39
     37.03
     72.58

withnlo_posGenW_noHEMcut
      0.00
   -128.74
   -531.51
   -586.78
   -508.76
   -507.10
   -499.50
   -458.65
   -539.52
   -511.20
   -822.96
   -558.29
   -575.70
   -867.01

withoutnlo_negGenW_applyHEMcut
      0.00
     45.95
    135.07
    205.38
     99.05
    112.

In [69]:
for cat in compare_hists.keys():
    print(cat)
    letssum = 0
    for i in range(len(compare_hists[cat]['WJets-1J'])):
        #print('%10.2f' % compare_hists[cat]['WJets-1J'][i])
        letssum = letssum+compare_hists[cat]['WJets-1J'][i]
    print('sum: ', letssum)
    print()

withoutnlo_posGenW_applyHEM
sum:  5704.863818858616

withnlo_posGenW_applyHEM
sum:  5179.013483712787

withnlo_allGenW_applyHEM
sum:  6462.916812865159

withoutnlo_allGenW_applyHEM
sum:  11810.374785611451

withnlo_posGenW_noHEMcut
sum:  -18764.110219522772

withoutnlo_negGenW_applyHEMcut
sum:  3472.099212046171

withnlo_negGenW_applyHEMcut
sum:  3154.8398243477773

withoutnlo_negGenW_noHEMcut
sum:  4362.51004060334

withnlo_negGenW_noHEMcut
sum:  3962.8306803877954

withoutnlo_posGenW_noHEMcut
sum:  7161.689593752289

withnlo_allGenW_removeHEMmetonCR
sum:  7505.169936783725



withnlo_posGenW_applyHEM
sum:  5179.013483712787

withnlo_negGenW_applyHEMcut
sum:  3154.8398243477773
    
withnlo_allGenW_applyHEM
sum:  6462.916812865159

withoutnlo_posGenW_applyHEM
sum:  5704.863818858616
    
withoutnlo_negGenW_applyHEMcut
sum:  3472.099212046171
    


In [5]:
vlabel = {'71': 'Fix sumw positive genW',
         '72': 'Fix sumw negative genW',
         '73': 'Fix sumw all genW'}
region = 'wmcr'
vari  = 'recoilphi'
fix_sumw = {}
systematic = 'nominal'

In [19]:
# With nlo ewk, positive genWeight, correct HEM cut (file V71)
version = 71
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'.scaled')
bkg_hists = hists['bkg']
fix_sumw['71']={
    'WJets-1J': bkg_hists[vari].integrate('region', region).integrate('systematic','nominal').values()[('WJets-1J',)],
}


In [ ]:
rebin_title = 'lets find issue'
ak15bin = [250, 345, 440, 535, 630, 725, 915, 1200]
rebkg_hists = bkg_hists[v].rebin(v, hist.Bin('newbin', rebin_title, [250, 345, 440, 535, 630, 725, 915, 1200]))
plot.plot1d(rebkg_hists.integrate('region', region).integrate('systematic',systematic),
        overlay="process",
        ax=ax,
        overflow='over',
        clear=False,
        stack=True,
        fill_opts=fill_opts,
        error_opts=error_opts
)

In [20]:
# With nlo ewk, negative genWeight, correct HEM cut (file V72)
version = 72
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'.scaled')
bkg_hists = hists['bkg']
fix_sumw['72']={
    'WJets-1J': bkg_hists[vari].integrate('region', region).integrate('systematic','nominal').values()[('WJets-1J',)],
}


In [21]:
# With nlo ewk, negative genWeight, correct HEM cut (file V73)
version = 73
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'.scaled')
bkg_hists = hists['bkg']
fix_sumw['73']={
    'WJets-1J': bkg_hists[vari].integrate('region', region).integrate('systematic','nominal').values()[('WJets-1J',)],
}


In [22]:
for cat in fix_sumw.keys():
    print(cat)
    for i in range(14):
        print('%10.2f' % fix_sumw[cat]['WJets-1J'][i])
    print()

73
      0.00
     46.17
    210.64
    205.59
    222.06
    208.50
    216.60
    159.42
    220.85
    210.25
    319.42
    187.48
    192.68
    358.75

71
      0.00
     68.27
    275.74
    304.91
    269.79
    262.81
    264.88
    243.22
    275.47
    237.95
    357.98
    247.65
    284.50
    427.62

72
      0.00
    -22.10
    -65.09
    -99.32
    -47.74
    -54.31
    -48.28
    -83.80
    -54.62
    -27.70
    -38.56
    -60.17
    -91.82
    -68.87



In [27]:
cat_name = {
    '71': 'Positive genW',
    '72': 'Negative genW',
    '73': 'All genW'
}
for cat in fix_sumw.keys():
    print(cat_name[cat])
    letssum = 0
    for i in range(len(fix_sumw[cat]['WJets-1J'])):
        #print('%10.2f' % compare_hists[cat]['WJets-1J'][i])
        letssum = letssum+fix_sumw[cat]['WJets-1J'][i]
    print('yield sum: %.2f' % letssum)
    print()

All genW
yield sum: 7505.17

Positive genW
yield sum: 9520.21

Negative genW
yield sum: -2015.04



In [6]:
##### DATA
# With nlo ewk, all genWeight, correct HEM cut (file V60)
version = 60
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'.scaled')
bkg_hists = hists['bkg']
dat_hists = hists['data']
vari = 'recoil'
region = 'tecr'
fix_sumw['60']={
    'data': dat_hists[vari].integrate('region', region).integrate('systematic','nominal').values()[('EGamma',)],
}


In [7]:
dat_hists[vari].integrate('region', region).integrate('systematic','nominal').sum(vari).values()

{('EGamma',): 8312.0}

In [8]:
bkg_hists[vari].integrate('region', region).integrate('systematic','nominal').integrate('process').sum(vari).values()

{(): 5965.401433692916}

In [61]:
8896+2407

11303

In [62]:
5205.29+225.46

5430.75

In [57]:
dat_hists['recoil'].values()

{('EGamma',
  'wecr',
  'nominal'): array([    0., 12316.,  9360.,  7149.,  5466.,  4112.,  3201.,  4373.,
         4292.,  2941.,   263.]),
 ('EGamma',
  'tecr',
  'nominal'): array([   0., 2045., 1510., 1119.,  870.,  623.,  481.,  676.,  596.,
         361.,   31.]),
 ('EGamma',
  'zecr',
  'nominal'): array([  0., 329., 240., 148., 121.,  82.,  51.,  70.,  66.,  31.,   0.]),
 ('EGamma',
  'gcr',
  'nominal'): array([    0., 30738., 21641., 15689., 11378.,  8512.,  6260.,  8376.,
         7760.,  4928.,   334.])}

In [69]:
bkg_hists.keys()

dict_keys(['sumw', 'template', 'cutflow', 'recoil', 'recoilphi', 'mT', 'eT_miss', 'eTphi_miss', 'nfj', 'ndflvj', 'ntighte', 'ntightmu', 'ntightpho', 'fj1pt', 'fj1phi', 'fj1eta', 'elept', 'elephi', 'eleeta', 'mupt', 'muphi', 'mueta', 'phopt', 'phophi', 'phoeta', 'dphi_fj_recoil', 'dphi_j_met', 'partWpt', 'partWpt2', 'genWeight_roi', 'nloewk_roi', 'TvsQCD'])

In [89]:
region = 'wecr'
for i in range(len(bkg_hists['cutflow'].integrate('region', region).integrate('process').values()[()])):
    a = bkg_hists['cutflow'].integrate('region', region).integrate('process').values()[()][i]
    print('%.2f' % a)
print()
for i in range(len(dat_hists['cutflow'].integrate('region', region).integrate('process').values()[()])):
    a = dat_hists['cutflow'].integrate('region', region).integrate('process').values()[()][i]
    print('%.1f' % a)

3997502500.68
3996974562.43
296988117.29
285852193.56
278374188.52
278309413.90
111725.54
109258.51
107202.23
107157.66
92112.95
87494.55
86459.90
76763.66
47542.35
0.00
0.00
0.00
0.00
0.00

1329168820.0
1327453226.0
643031738.0
602526956.0
602526956.0
602526956.0
156798.0
150785.0
147139.0
147044.0
118827.0
109633.0
103973.0
89808.0
53485.0
0.0
0.0
0.0
0.0
0.0


In [86]:
53485/43591.11

1.2269703616173113

In [87]:
48723.0/41698.56

1.1684576158025601

In [38]:
##### DATA
# manymanyplopt.scaled version From Taiwoo
version = 62
year = 2018
print(path+'manyplot_'+str(year)+'_V'+str(version)+'.scaled')
hists = load(path+'manyplot_'+str(year)+'_V'+str(version)+'.scaled')
bkg_hists = hists['bkg']
dat_hists = hists['data']
region = 'wecr'
vari = 'recoil'

#for i in range(len(bkg_hists['cutflow'].integrate('region', region).integrate('process').values()[()])):
#    a = bkg_hists['cutflow'].integrate('region', region).integrate('process').values()[()][i]
#    print('%.2f' % a)
#print()
for i in range(len(dat_hists['cutflow'].integrate('region', region).integrate('process').values()[()])):
    a = dat_hists['cutflow'].integrate('region', region).integrate('process').values()[()][i]
    print('%.1f' % a)

/Users/hongjieun/MonoTop/notebook_plotting/check_events/manyplot_2018_V62.scaled
1329168820.0
1327453226.0
643031738.0
602526956.0
602526956.0
602526956.0
156798.0
150785.0
147139.0
147044.0
118827.0
109633.0
103973.0
89808.0
53485.0
0.0
0.0
0.0
0.0
0.0


In [33]:
dat_hists[vari].integrate('systematic','nominal').values()

{('EGamma',
  'wecr'): array([    0., 11826.,  8975.,  6824.,  5258.,  3939.,  3066.,  4190.,
         4135.,  2825.,   249.]),
 ('EGamma',
  'tecr'): array([   0., 1933., 1428., 1049.,  837.,  596.,  459.,  649.,  572.,
         347.,   28.]),
 ('EGamma',
  'zecr'): array([  0., 320., 234., 141., 115.,  79.,  49.,  69.,  63.,  30.,   0.]),
 ('EGamma',
  'gcr'): array([    0., 29574., 20800., 15113., 10966.,  8213.,  6033.,  8060.,
         7466.,  4735.,   322.])}